site/utils/apiUtils/createDisruptionValidations.ts

```diff
@@ -0,0 +1,278 @@
+    errors: ErrorInfo[],
+    errorId: string,
+) => {
+    let error: ErrorInfo = {
+        id: errorId,
+        errorMessage: "Please choose a Reason from the dropdown",
+    };
+    if (!requireFieldCheck(disruptionReason)) {
+        errors.push(error);
+    } else {
+        error = {
+            id: errorId,
+            errorMessage: "Invalid value provided for Reason for Disruption",
+        };
+
+        if (
+            Object.values(MiscellaneousReason).filter((value: string) => value === disruptionReason).length == 0 &&
+            Object.values(EnvironmentReason).filter((value: string) => value === disruptionReason).length == 0 &&
+            Object.values(PersonnelReason).filter((value: string) => value === disruptionReason).length == 0 &&
+            Object.values(EquipmentReason).filter((value: string) => value === disruptionReason).length == 0
+        ) {
+            errors.push(error);
+        }
+    }
+};
+
+export const validateDateTime = (
+    date: Date | null,
+    time: string,
+    errors: ErrorInfo[],
+    errorId: string,
+    dateType: string,
+): boolean => {
+    let isValid = false;
+
+    const timeCheck = requireFieldCheck(time);
+    if (!date) {
+        errors.push({
+            id: errorId,
+            errorMessage: `No ${dateType} date selected. Please select a valid ${dateType} date`,
+        });
+    }
+
+    if (!timeCheck) {
+        errors.push({
+            id: errorId,
+            errorMessage: `No ${dateType} time entered. Please select a valid ${dateType} time`,
+        });
+    }
+
+    if (date && timeCheck) {
+        const jsDate = dayjs(date, CD_DATE_FORMAT, "en-gb", true);
+        const isValidTime = validateTime(time, dateType, errors, errorId);
+
+        if (!jsDate.isValid()) {
+            errors.push({
+                id: errorId,
+                errorMessage: `Invalid ${dateType} Date value submitted. Please select a valid ${dateType} date`,
+            });
+        } else {
+            // Validate that the start time input is valid and of time format
+            const isFutureDate = jsDate.hour(+time.slice(0, 2)).minute(+time.slice(2, 4)).isAfter(dayjs());
+
+            if (isValidTime && !isFutureDate) {
+                errors.push({
+                    id: errorId,
+                    errorMessage: `The ${dateType} date and time should be past the current date and time`,
+                });
+            } else {
+                isValid = isValidTime;
+            }
+        }
+    }
+
+    return isValid;
+};
+
+export const validateTime = (time: string, dateType: string, errors: ErrorInfo[], errorId: string): boolean => {
+    let isValid = false;
+
+    if (time.length != 4) {
+        errors.push({
+            id: errorId,
+            errorMessage: `Value for ${dateType} time is of invalid format. Please enter the time in 24hr format. For example 0900 is 9am, 1730 is 5:30pm`,
+        });
+    } else if (isNaN(+time)) {
```

Comment:
I think we should combine these error checks into one. We shouldnt be telling them its not a number, because to a user, 'three' is a number. Its simple enough to just tell them its not the correct format, and they need to use hhmm.